src/test/java/uk/gov/hmcts/reform/idam/web/UserControllerTest.java

```diff
@@ -512,4 +514,16 @@ public void selfRegister_shouldPopulateTheModelWithTheUsersDetailsIfCalledWithAV
             .andExpect(model().attribute("email", nullValue()))
             .andExpect(view().name(SELF_REGISTER_VIEW_NAME));
     }
+
+    /**
+     * @verifies return null if redirecturi or clientid are empty
+     * @see UserController#buildRegistrationLink(ActivationResult)
+     */
+    @Test
+    public void buildRegistrationLink_shouldReturnNullIfRedirecturiOrClientidAreEmpty() throws Exception {
+        UserController userController = new UserController();
+        assertNull(userController.buildRegistrationLink(new ActivationResult()));
+        assertNull(userController.buildRegistrationLink(new ActivationResult().redirectUri(GOOGLE_WEB_ADDRESS)));
+        assertNull(userController.buildRegistrationLink(new ActivationResult().clientId(CLIENT_ID)));
+    }
```

Comment:
not sure this is the best way to test a controller

if you follow the pattern in use in this test file you can use mockMvc and test whatever is using the `buildRegistrationLink` method

Reply:
I think this is better than using mocks.

Reply:
@tbamido let me know if you're okay with it staying as a normal unit test